viewer/src/glsl/base/updateFragmentColor.glsl

```diff
@@ -2,24 +2,50 @@
 #pragma glslify: hsv2rgb = require('../color/hsv2rgb.glsl')
 #pragma glslify: packIntToColor = require('../color/packIntToColor.glsl')
 
-#ifdef COGNITE_COLOR_BY_TREE_INDEX
+const int RenderTypeColor = 1;
+const int RenderTypeNormal = 2;
+const int RenderTypeTreeIndex = 3;
+const int RenderTypePackColorAndNormal = 4;
 
-void updateFragmentColor(vec3 color, float treeIndex, vec3 normal) {
-    color = packIntToColor(treeIndex);
-    float amplitude = max(0.0, dot(normal, vec3(0.0, 0.0, 1.0)));
-    gl_FragColor = vec4(color * (0.4 + 0.6 * amplitude), 1.0);
-}
+uniform int renderType;
 
-#else 
+vec3 packNormalToRGB( const in vec3 normal ) {
+	return normalize( normal ) * 0.5 + 0.5;
+}
 
 void updateFragmentColor(vec3 color, float treeIndex, vec3 normal) {
-    vec3 hsv = rgb2hsv(color);
-    hsv.z = min(0.6 * hsv.z + 0.4, 1.0);
-    color = hsv2rgb(hsv);
-    float amplitude = max(0.0, dot(normal, vec3(0.0, 0.0, 1.0)));
-    gl_FragColor = vec4(color * (0.4 + 0.6 * amplitude), 1.0);
+    if (renderType == RenderTypeColor) {
+        vec3 hsv = rgb2hsv(color);
+        float h = hsv.x;
+        hsv.z = min(0.6 * hsv.z + 0.4, 1.0);
+        color = hsv2rgb(hsv);
+        float amplitude = max(0.0, dot(normal, vec3(0.0, 0.0, 1.0)));
+        gl_FragColor = vec4(color * (0.4 + 0.6 * amplitude), h);
+    } else if (renderType == RenderTypePackColorAndNormal) {
+        vec3 hsv = rgb2hsv(color);
+        float a = 0.0;
+        if (hsv.y > 0.01) {
+            if (hsv.z > 0.5) {
+                // light color
+                a = (0.55 + hsv.x * 0.45);
+            } else {
+                // dark color
+                a = (0.1 + hsv.x * 0.45);
+            }
+        } else {
+            // no saturation - grayscale
+            a = hsv.z * 0.09;
+        }
+        gl_FragColor = vec4(normal.rgb, a);
```

Comment:
Won't normal.rgb derp on negative components?  Need to use packNormalToRGB() as you do below?

Reply:
Good point! Fixed now.